pocs/tests/serial_handlers/protocol_test.py

```diff
@@ -0,0 +1,243 @@
+        # Read-write property values
+        self._break_condition = False
+        self._rts = False
+        self._dtr = False
+        # Read-only property values
+        self._name = False
+        self._rts = False
+        self._ctx = False
+        self._dsr = False
+        self._ri = False
+        self._ri = False
+        self._ri = False
+        self._ri = False
+        self.logger = None
+        self.hooks = the_hook_creator(self, *args, **kwargs)
+        super().__init__(*args, **kwargs)
+        self.hooks.created(self)
+
+    def open(self):
+        """Open port.
+        
+        Raises:
+            SerialException if the port cannot be opened.
+        """
+        self.hooks.open()
+
+    def close(self):
+        """Close port"""
+        self.hooks.close()
+
+    def read(self, size=1):
+        """Read size bytes from the serial port."""
+        if not self._isOpen:
+            raise portNotOpenError
+        return self.hooks.read(size)
+
+    def write(self, data):
+        """Write the given data to the serial port."""
+        if not self._isOpen:
+            raise portNotOpenError
+        return self.hooks.write(data)
+
+    #  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+
+    @property
+    def in_waiting(self):
+        """Return the number of characters currently in the input buffer."""
+        if not self._isOpen:
+            raise portNotOpenError
+        if self.logger:
+            # set this one to debug as the function could be called often...
+            self.logger.debug('WARNING: in_waiting returns dummy value')
+        return 0  # hmmm, see comment in read()
+
+    def reset_input_buffer(self):
+        """Flush input buffer, discarding all it’s contents."""
+        if not self._isOpen:
+            raise portNotOpenError
+
+    def reset_output_buffer(self):
+        """Clear output buffer.
+        
+        Clear output buffer, aborting the current output and discarding
+        all that is in the buffer.
+        """
+        if not self._isOpen:
+            raise portNotOpenError
+        if self.logger:
+            self.logger.info('ignored flushOutput')
+
+    def send_break(self, duration=0.25):
+        """Send break condition. Timed, returns to idle state after given
+        duration."""
+        if not self._isOpen:
+            raise portNotOpenError
+        if self.logger:
+            self.logger.info('ignored sendBreak({!r})'.format(duration))
+
+    @property
+    def break_condition(self):
+        """Get break_condition."""
+        return self.__break_condition
+
+    @break_condition.setter
+    def break_condition(self, value):
+        """Set break_condition: Controls TXD. When active, to transmitting is
+        possible."""
+        self.__break_condition = value
+        if self.logger:
+            self.logger.info('ignored setBreak({!r})'.format(level))
+
+    def setRTS(self, level=True):
```

Comment:
All these method names are deprecated in favor of properties. Should we still be using?